tileview/src/main/java/com/qozix/tileview/tiles/TileCanvasViewGroup.java

```diff
@@ -366,7 +364,7 @@ private void markTilesAsPrevious() {
   private void beginRenderTask() {
     // if visible columns and rows are same as previously computed, fast-fail
     boolean changed = mDetailLevelToRender.computeCurrentState();
-    if( !changed && mDetailLevelToRender.equals( mLastRenderedDetailLevel ) ) {
+    if( !changed && mTilesInCurrentViewport.size() > 0 ) {
```

Comment:
is this logic right?  if there is no change, and there are no tiles, fast fail...?  shouldn't that be "or" rather than "and"?

Reply:
Sorry for the late answer. 
The logic says : if there are no tiles, do not fast fail.
This is sadly the only fix i found. Like explained, i had to find a way to restrain the fast fail occurrences.

Reply:
hm, i can't say i get it at this point, but i trust you, and we're pretty much in maintenance mode at this point, so i'm going to merge and push this out to jcenter.

thanks @peterLaurence!